src/wp-includes/option.php

```diff
@@ -253,6 +253,107 @@ function get_option( $option, $default_value = false ) {
 	return apply_filters( "option_{$option}", maybe_unserialize( $value ), $option );
 }
 
+/**
+ * Primes specific options into the cache with a single database query.
+ *
+ * Only options that do not already exist in cache will be primed.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options An array of option names to be primed.
+ */
+function prime_options( $options ) {
+	$alloptions     = wp_load_alloptions();
+	$cached_options = wp_cache_get_multiple( $options, 'options' );
+
+	// Filter options that are not in the cache.
+	$options_to_prime = array();
+	foreach ( $options as $option ) {
+		if ( ( ! isset( $cached_options[ $option ] ) || ! $cached_options[ $option ] ) && ! isset( $alloptions[ $option ] ) ) {
+			$options_to_prime[] = $option;
+		}
+	}
+
+	if ( ! empty( $options_to_prime ) ) {
+		global $wpdb;
+		$results = $wpdb->get_results(
+			$wpdb->prepare(
+				sprintf(
+					"SELECT option_name, option_value FROM $wpdb->options WHERE option_name IN (%s)",
+					implode( ',', array_fill( 0, count( $options_to_prime ), '%s' ) )
+				),
+				$options_to_prime
+			)
+		);
+
+		$options_not_found = $options_to_prime;
+		foreach ( $results as $result ) {
+			wp_cache_set( $result->option_name, maybe_unserialize( $result->option_value ), 'options' );
+			unset( $options_not_found[ array_search( $result->option_name, $options_not_found, true ) ] );
+		}
+
+		$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+		// Make a copy to check later if any change was made.
+		$new_notoptions = $notoptions;
+
+		// Add the options that were not found to the cache.
+		if ( ! is_array( $new_notoptions ) ) {
+			$new_notoptions = array();
+		}
+		foreach ( $options_not_found as $option_name ) {
+			if ( ! isset( $new_notoptions[ $option_name ] ) ) {
+				$new_notoptions[ $option_name ] = true;
+			}
+		}
+
+		// Only update the cache if it was modified.
+		if ( $new_notoptions !== $notoptions ) {
+			wp_cache_set( 'notoptions', $new_notoptions, 'options' );
+		}
+	}
+}
+
+/**
+ * Wrapper function to prime all options registered with a specific option group.
```

Comment:
```suggestion
 * Primes all options registered with a specific option group.
```
Function and method docblocks should start with a third-person singular verb.